test/e2e/utils/helpers.go

```diff
@@ -160,48 +160,16 @@ func IsScyllaClusterRolledOut(sc *scyllav1.ScyllaCluster) (bool, error) {
 	return true, nil
 }
 
-func WaitForScyllaClusterState(ctx context.Context, client scyllav1client.ScyllaV1Interface, namespace string, name string, conditions ...func(sc *scyllav1.ScyllaCluster) (bool, error)) (*scyllav1.ScyllaCluster, error) {
-	fieldSelector := fields.OneTermEqualSelector("metadata.name", name).String()
-	lw := &cache.ListWatch{
-		ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
-			options.FieldSelector = fieldSelector
-			return client.ScyllaClusters(namespace).List(ctx, options)
-		},
-		WatchFunc: func(options metav1.ListOptions) (i watch.Interface, e error) {
-			options.FieldSelector = fieldSelector
-			return client.ScyllaClusters(namespace).Watch(ctx, options)
-		},
-	}
-
-	event, err := watchtools.UntilWithSync(ctx, lw, &scyllav1.ScyllaCluster{}, nil, func(event watch.Event) (bool, error) {
-		switch event.Type {
-		case watch.Added, watch.Modified:
-			for _, condition := range conditions {
-				done, err := condition(event.Object.(*scyllav1.ScyllaCluster))
-				if err != nil {
-					return false, err
-				}
-				if !done {
-					return false, nil
-				}
-			}
-			return true, nil
-		default:
-			return true, fmt.Errorf("unexpected event: %#v", event)
-		}
-	})
-	if err != nil {
-		return nil, err
-	}
-
-	return event.Object.(*scyllav1.ScyllaCluster), nil
+type listerWatcher[ListPtr runtime.Object] interface {
```

Comment:
```suggestion
type listerWatcher[ObjectList runtime.Object] interface {
```
for consistency

Reply:
made it `ListObject`. (`ObjectList` wouldn't be a pointer as Lists are called the actual non-ptr types, not the interfaces)